internal/go_repository_config.bzl

```diff
@@ -26,33 +26,38 @@ def _go_repository_config_impl(ctx):
         for label in _find_macro_file_labels(ctx, ctx.attr.config):
             ctx.path(label)
 
-    env = read_cache_env(ctx, str(ctx.path(Label("@bazel_gazelle_go_repository_cache//:go.env"))))
-
-    generate_repo_config = str(ctx.path(Label("@bazel_gazelle_go_repository_tools//:bin/generate_repo_config{}".format(executable_extension(ctx)))))
-    list_repos_args = [
-        "-config_source=" + str(config_path),
-        "-config_dest=" + str(ctx.path("WORKSPACE")),
-    ]
-    result = env_execute(
-        ctx,
-        [generate_repo_config] + list_repos_args,
-        environment = env,
+    if config_path:
+        env = read_cache_env(ctx, str(ctx.path(Label("@bazel_gazelle_go_repository_cache//:go.env"))))
+        generate_repo_config = str(ctx.path(Label("@bazel_gazelle_go_repository_tools//:bin/generate_repo_config{}".format(executable_extension(ctx)))))
+        list_repos_args = [
+            "-config_source=" + str(config_path),
+            "-config_dest=" + str(ctx.path("WORKSPACE")),
+        ]
+        result = env_execute(
+            ctx,
+            [generate_repo_config] + list_repos_args,
+            environment = env,
+        )
+        if result.return_code:
+            print("generate_repo_config: " + result.stderr)
```

Comment:
I think this should be `fail` instead of `print`. If we can't create the configuration file, that would be a fatal error, right?